### ISSUE TITLE
Ghostery is adding about 1,6 MB content to each mail I send via OWA

### ISSUE THREAD

**Issue by Misiu** [ghostery/ghostery-extension#381]
### Description

I've installed Ghostery for Chrome and noticed that when I send emails via Outlook Web App my emails got larger and some recipients servers are blocking my emails because of invalid content in them.

I've send two emails with the exact same content, first without Ghostery disabled second with enabled.

Below are the differences:
Ghostery disabled:
email size:  39 KB
email length: 748 lines of HTML

Ghostery enabled:
email size:  **1.67 MB**
email length: **57950** lines of HTML

### Expected Behavior 

Emails that are sent via OWA shouldn't have any additional content related to Ghostery.

### Actual Behavior 

Mails got larger and some recipients servers are blocking them, probably because of disallowed keywords (a lot of porn sites references):

![ghostery](https://user-images.githubusercontent.com/1741838/57434707-dbeead80-723b-11e9-9926-15d1f31033a1.png)

### Versions

+ Browser: Google Chrome  74.0.3729.131
+ OS: Windows 10

**Comment by christophertino** [comment]
Looks like Ad Blocker rules.  @remusao any ideas?

**Comment by sammacbeth** [comment]
These are Adblocker filter rules. We have changed the way we inject these so this should get fixed once browser-core is updated to 1.36.

**Comment by Misiu** [comment]
@christophertino @sammacbeth thanks for looking into this.
When will browser-core get updated to 1.36?
Currently, I had to disable Ghostery to be able to use OWA.

**Comment by Misiu** [comment]
@sammacbeth any updates?

**Comment by Shoe-Pi** [comment]
Just to chip in, I'm haivng the exact same issue right now, and much of my work email was getting bounced by the servers for malware before disabling it.

> These are Adblocker filter rules. We have changed the way we inject these so this should get fixed once browser-core is updated to 1.36.

Why the heck is Ghostery even injecting things into my emails?

**Comment by remusao** [comment]
@Shoe-Pi @Misiu,

Sorry for the late reply and thanks a lot for the detailed feedback on this one. I tried to reproduce the issue by sending emails with Ghostery adblocker enabled from `https://outlook.live.com` (is that correct?), but could not reproduce this so far.

What I assume is happening here is this:

* Ghostery's adblocker will do two things to block ads on pages:
  1. Block network requests and prevent them from leaving your browser (this will prevent most ads from loading)
  2. Some ads cannot be blocked that way and need to be hidden on the pages instead: this is why Ghostery will inject custom stylesheets (a.k.a.: CSS) in the pages.
* For some reason, on OWA, the custom injected style is also kept as part of the email; that should indeed not happen.

We already have a fix for that which will be released in next version of Ghostery; instead of injecting stylesheets directly in the pages (i.e.: the DOM), we use another API from the browser so that the CSS is not visible from the page, which should solve the issue. In the meanwhile, you could disable the adblocker while on this particular domain.

**Comment by Misiu** [comment]
@remusao I've already disabled Ghostery for certain domains, but this is a temporary fix. BTW I had to add a domain without subfolder, for example, OWA in my case is available at `http://m.abc.pl/owa/` I had to add `m.abc.pl` because it didn't work with full URL (is that intended or is this a bug?)

I use old Exchange Server, maybe new OWA filters outgoing HTML and removes CSS.
Looking forward to that next release, hopefully, this issue will be solved.
